### ISSUE TITLE
Update CMake build

### ISSUE THREAD

**Issue by zneak** [zneak/fcd#28]
LLVM 3.9 made it to public package repositories some time ago. I've just updated the Makefile, but CMake remains behind.

@Trass3r?

**Comment by Trass3r** [comment]
Ideally everything should converge to 1 generic build system.
How is the experience with a CMake generated xcode project?

**Comment by zneak** [comment]
It takes some fighting. I stopped at the third problem but I know that I'd hit at least a 4th one:

* First, `xcode-select` needs to point to an Xcode installation, and not to standalone developer tools, before running `cmake -G Xcode`. (This is arguably on my end, but still not a great experience.)
* Second, Capstone literally [can't be installed](https://en.wikipedia.org/wiki/System_Integrity_Protection) in /usr on macOS, a problem that I "solved" by replacing the path with my local installation's path.
* Third, Xcode fails to build the x86 emulator assembly file, presumably because it's not where it expects it to be. The command that builds the .asm file launches from the fcd source root directory and the bytecode file is in root/build (build being the directory from which I ran cmake). The same command to build it works (from the command line) if I run it from the build directory instead.
* Inferred fourth problem: the project uses the Linux asm template for the emulator `.incbin` directive, but the osx template is necessary. Aside from the different section/visibility for the symbols (which is merely nice to have), the macOS C ABI "mangles" names by prefixing them with an underscore. Linux doesn't do that and has a `fcd_emulator_start_x86` symbol, but a C compiler on macOS can't find this symbol because it needs to be called `_fcd_emulator_start_x86`.

Aside that it doesn't build, I'm pretty comfortable using a "native" Xcode project. It's not entirely clear that managing files from the CMake-generated project would be straightforward, since it creates an entirely different file hierarchy (every .h in a single group and every .cpp in another group, whereas the native Xcode project mixes .h and .cpp files and groups them by component). I know that I could live with it as an occasional contributor, but as the project's owner, I don't know if I want to do that tradeoff.
